sql/importer: unskip TestImportWorkerFailure in more configs

Different errors encountered (some are w/ and some are w/o closed #107761):

`COCKROACH_TEST_TENANT=true`:
```
    import_stmt_test.go:5474: pq: addsstable [/Tenant/10/Table/104/1/2/0,/Tenant/10/Table/104/1/17/0/NULL): tenant-side-limiter pool closed: shutting down
```
```
    import_stmt_test.go:5474: pq: updating job details after publishing schemas: job 888134426637434881: select-job: grpc: grpc: the client connection is closing [code 1/Canceled]
```
```
    import_stmt_test.go:5474: pq: failed to resolve n2: unable to look up descriptor for n2: non existent SQL instance
```
```
* INFO: slow quiesce. stack traces:
```

----

`COCKROACH_TEST_TENANT=false`:
  - `--cpus=48`
```
    jobs_verification.go:88: error scanning '&{<nil> 0xc007344400}': sql: no rows in result set
```
  - `--cpus=32`
```
* INFO: slow quiesce. stack traces:
```
 - `--cpus=32`
```
    import_stmt_test.go:5479: query 'SELECT * FROM t ORDER BY i': expected:
        0
        1
        2
        3
        4
        5
        6
        7
        8
        9
        10
        11
        12
        13
        14
        15
        16
        17
        18
        19

        got:
        1
        2
        4
        5
        7
        8
        10
        11
        13
        14
        16
        17
        19
```

<details>
<summary>original issue</summary>

sql/importer.TestImportWorkerFailure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11038168?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11038168?buildTab=artifacts#/) on master @ [41229997ab1a6bef8e7a7978ee65d88aab6f008e](https://github.com/cockroachdb/cockroach/commits/41229997ab1a6bef8e7a7978ee65d88aab6f008e):


```
=== RUN   TestImportWorkerFailure
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/7b0bc1d41114c8beed24a57e2afd8437/logTestImportWorkerFailure1576023233
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:104: 
        Test server was configured to route SQL queries to a secondary tenant (virtual cluster).
        If you are only seeing a test failure when this message appears, there may be a problem
        specific to cluster virtualization or multi-tenancy.
        
        To investigate, consider using "COCKROACH_TEST_TENANT=true" to force-enable just
        the secondary tenant in all runs (or, alternatively, "false" to force-disable), or use
        "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false" to disable all random test variables altogether.
    testcluster.go:415: 
        Test server was configured to route SQL queries to a secondary tenant (virtual cluster).
        If you are only seeing a test failure when this message appears, there may be a problem
        specific to cluster virtualization or multi-tenancy.
        
        To investigate, consider using "COCKROACH_TEST_TENANT=true" to force-enable just
        the secondary tenant in all runs (or, alternatively, "false" to force-disable), or use
        "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false" to disable all random test variables altogether.
    import_stmt_test.go:5474: pq: addsstable [/Tenant/10/Table/104/1/2/0,/Tenant/10/Table/104/1/17/0/NULL): tenant-side-limiter pool closed: shutting down
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7b0bc1d41114c8beed24a57e2afd8437/logTestImportWorkerFailure1576023233
--- FAIL: TestImportWorkerFailure (76.71s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102839 sql/importer: TestImportWorkerFailure failed [C-test-failure O-robot T-sql-queries branch-release-23.1 skipped-test]
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportWorkerFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

</details>


Jira issue: CRDB-30053

--- a/pkg/sql/importer/import_stmt_test.go
+++ b/pkg/sql/importer/import_stmt_test.go
@@ -5321,9 +5321,8 @@ func TestImportWorkerFailure(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.UnderStressWithIssue(t, 108547, "flaky test")
-	skip.UnderDeadlockWithIssue(t, 108547, "flaky test")
-	skip.UnderRaceWithIssue(t, 108547, "flaky test")
+	skip.UnderDeadlock(t, "test is flaky under deadlock")
+	skip.UnderStressRace(t, "test is flaky under stressrace")
 
 	allowResponse := make(chan struct{})
 	params := base.TestClusterArgs{}

